src/bundles/ar/AR.ts

```diff
@@ -0,0 +1,272 @@
+}
+
+/**
+ * Sets the right toggle.
+ *
+ * @param text Label on toggle.
+ * @param callback Function to call when toggle is clicked.
+ */
+export function setRightToggle(text: string, callback: () => void) {
+  const moduleState = getModuleState();
+  if (!moduleState) return;
+  moduleState.overlay.toggleRight = new Toggle(text, callback);
+}
+
+/**
+ * Resets and hides the left toggle.
+ */
+export function removeLeftToggle() {
+  const moduleState = getModuleState();
+  if (!moduleState) return;
+  moduleState.overlay.toggleLeft = undefined;
+}
+
+/**
+ * Resets and hides the center toggle.
+ */
+export function removeCenterToggle() {
+  const moduleState = getModuleState();
+  if (!moduleState) return;
+  moduleState.overlay.toggleCenter = undefined;
+}
+
+/**
+ * Resets and hides the right toggle.
+ */
+export function removeRightToggle() {
+  const moduleState = getModuleState();
+  if (!moduleState) return;
+  moduleState.overlay.toggleRight = undefined;
+}
+
+// Objects
+
+/**
+ * Creates an instance of Vector3.
+ *
+ * @param x Value of x-axis.
+ * @param y Value of y-axis.
+ * @param z Value of z-axis.
+ * @returns Vector3 created from specified values.
+ */
+export function createVector3(x: number, y: number, z: number): Vector3 {
+  return new Vector3(x, y, z);
+}
+
+/**
+ * Adds the specified object to the augmented world.
+ *
+ * @param object ARObject to add. (E.g. cube, sphere, etc..)
+ */
+export function addARObject(object: ARObject) {
+  const moduleState = getModuleState();
+  if (!moduleState) return;
+  if (moduleState.arObjects.find((item) => item.id === object.id)) {
+    return; // Already in array
+  }
+  if (object.onSelect) {
+    moduleState.clickCallbacks.set(object.id, () => {
+      object.onSelect?.(object);
+    });
+  }
+  const newArray = Object.assign([], moduleState.arObjects);
+  newArray.push(object);
+  moduleState.arObjects = newArray;
+  callARCallback();
+}
+
+/**
+ * Removes the specified object from the augmented world.
+ *
+ * @param object ARObject to remove.
+ */
+export function removeARObject(object: ARObject) {
+  const moduleState = getModuleState();
+  if (!moduleState) return;
+  moduleState.arObjects = moduleState.arObjects.filter(
+    (item) => item.id !== object.id,
+  );
+  callARCallback();
+}
+
+/**
+ * Obtains the current ARObjects.
+ */
+export function getARObjects(): ARObject[] {
+  const moduleState = getModuleState();
+  if (!moduleState) return [];
+  return moduleState.arObjects;
+}
+
+/**
+ * Sets AR objects from json.
+ */
+export function setAsARObjects(json: any) {
+  const moduleState = getModuleState();
+  if (!moduleState) return;
+  if (!(json instanceof Object)) return;
+  const objects: ARObject[] = [];
+  const keys = Object.keys(json);
+  keys.forEach((key) => {
+    const item = json[key];
+    const parsedObject = ARObject.fromObject(item);
+    if (parsedObject) {
+      objects.push(parsedObject);
+    }
+  });
```

Comment:
This could be simplified using Object.entries(json)

Reply:
The ARObject.fromObject static function here is custom defined. I don't think this world work

Reply:
I meant we should use Object.values() to replace Object.keys() so the line 
  const item = json[key];
  could be removed. The rest of the code should stay the same.